### PR TITLE
add post comment feature

### PR DESCRIPTION
**Pull request by karobarari**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -176,4 +174,61 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid input");
+      });
+  });
+});
```
Comment (nickkagame) [inline __tests__/app.test.js:262]: Hi Karo, we just need a few more tests.  404 non existent article and 400 bad article request

File: app/app.controllers.js [hunk 2]
```diff
@@ -45,8 +46,20 @@ exports.getCommentsByArticle = (req, res, next) => {
...
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+exports.postComment = (req, res, next) => {
+  const { body, params } = req;
+
+  const passedComment = body;
+  passedComment.article_id = params.article_id;
```
Comment (nickkagame) [inline app/app.controllers.js:59]: this is puzzling.  What are you trying to do here?

**Review by nickkagame — COMMENTED** [review]
Hi Karo!  We just need a few more tests then we'll be golden!  See my comments for details. Good work.

File: app/app.controllers.js (hunk 2 above)
Comment (karobarari) [reply app/app.controllers.js:59]: I was trying to pass the body only 1 object parameter

File: __tests__/app.test.js [hunk 3]
```diff
@@ -176,4 +174,88 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found!");
+      });
+  });
+  test("status:404, responds with an error message when passed a non existent article id", () => {
+    const newComment = {
+      username: "Karo",
```
Comment (nickkagame) [inline __tests__/app.test.js]: You'd have to update the username to be one that exists in order to be sure it's the article and not the user that is not found

File: app/app.controllers.js [hunk 4]
```diff
@@ -45,8 +46,20 @@ exports.getCommentsByArticle = (req, res, next) => {
...
+  createComment(passedComment)
+    .then((postedCm) => {
+      res.status(201).send(postedCm);
```
Comment (nickkagame) [inline app/app.controllers.js]: please send as an object

**Review by nickkagame — APPROVED** [review]
hi Karo, I'll come over and talk to you about some feedback points in person, but for now you can move on.

**Pull request merged**